OAuth2/AuthorizationRoot.cs

```diff
@@ -69,7 +69,8 @@ public virtual IEnumerable<IClient> Clients
         /// </summary>        
         protected virtual IEnumerable<Type> GetClientTypes()
         {
-            return Assembly.GetExecutingAssembly().GetTypes().Where(typeof (IClient).IsAssignableFrom);
+          var clientType = typeof(IClient);
+          return AppDomain.CurrentDomain.GetAssemblies().SelectMany(s => s.GetTypes()).Where(p => clientType.IsAssignableFrom(p));
```

Comment:
I don't see the benefit of creating a client type variable when it would be shorter if you used the previous where clause. Also have you noticed any performance issues with this (The app domain might have hundreds/thousands of assemblies)?